### ISSUE TITLE
Build failure with GHC 8.4

### ISSUE THREAD

**Issue by DanBurton** [bokesan/IntervalMap#5]
Add semigroup instances.
See also: https://ghc.haskell.org/trac/ghc/wiki/Migration/8.4#SemigroupMonoidsuperclasses

```
Building library for IntervalMap-0.5.3.1..
[1 of 9] Compiling Data.IntervalMap.Interval ( Data/IntervalMap/Interval.hs, dist/build/Data/IntervalMap/Interval.o )
[2 of 9] Compiling Data.IntervalMap.Generic.Interval ( Data/IntervalMap/Generic/Interval.hs, dist/build/Data/IntervalMap/Generic/Interval.o )
[3 of 9] Compiling Data.IntervalMap.Generic.Base ( Data/IntervalMap/Generic/Base.hs, dist/build/Data/IntervalMap/Generic/Base.o )

Data/IntervalMap/Generic/Base.hs:254:10: error:
    • Could not deduce (Semigroup (IntervalMap i v))
        arising from the superclasses of an instance declaration
      from the context: (Interval i k, Ord i)
        bound by the instance declaration
        at Data/IntervalMap/Generic/Base.hs:254:10-58
    • In the instance declaration for ‘Monoid (IntervalMap i v)’
    |
254 | instance (Interval i k, Ord i) => Monoid (IntervalMap i v) where
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by bokesan** [comment]
Thanks for the report. Working on the fix. I wish I knew how important ghc 7.x compatibility is.

**Comment by bokesan** [comment]
Fixed in version 0.6.0.0
